### PR TITLE
feat(next-international): add `rewriteDefault` strategy

### PR DESCRIPTION
**Pull request by zackrw**

### CODE REVIEW

**Comment by QuiiBz** [comment]
Thanks for the PR. Could you give some insights about why this feature would be useful? Is it something you need?

File: packages/next-international/src/app/middleware/index.ts [hunk 1]
```diff
@@ -25,6 +25,15 @@ export function createI18nMiddleware<Locales extends readonly string[]>(
...
+      } else if (strategy === 'rewriteDefault') {
+        if (locale === defaultLocale) {
+          const response = NextResponse.rewrite(mappedUrl);
+          return addLocaleToResponse(response, locale);
+        }
+        else {
+          const response = NextResponse.redirect(mappedUrl);
+          return addLocaleToResponse(response, locale);
+        }        
       } else {
```
Comment (QuiiBz) [inline packages/next-international/src/app/middleware/index.ts:37]: We can simplify this quite a bit:

```suggestion
      if (strategy === 'rewrite' || (strategy === 'rewriteDefault' && locale === defaultLocale)) {
        const response = NextResponse.rewrite(mappedUrl);
        return addLocaleToResponse(response, locale);      
      } else {
```

File: packages/next-international/src/types.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export type I18nMiddlewareConfig<Locales extends readonly string[]> = {
    *
    * @default redirect
    */
-  urlMappingStrategy?: 'redirect' | 'rewrite';
+  urlMappingStrategy?: 'redirect' | 'rewrite' | 'rewriteDefault';
```
Comment (QuiiBz) [inline packages/next-international/src/types.ts:55]: Can we add `rewriteDefault` to the JSDoc above to explain what it does?

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for the insights, I think it's a great feature to have.

Let's change the target branch to `next`, where we're working on the stable v1 release: https://github.com/QuiiBz/next-international/pull/126 (I'm targeting mid-september)

We will have a proper documentation website so we can move the docs for `rewriteDefault` from the README [to here](https://next-international.vercel.app/docs/app-middleware-configuration#rewrite-the-url-to-hide-the-locale).

**Review by zackrw — COMMENTED** [review]
Looks good to me. Thanks for the comments.

**Review by QuiiBz — APPROVED** [review]
Thanks! I've just added some documentation and fixed the failing lint job.

**Pull request merged**